NuRadioMC/simulation/simulation.py

```diff
@@ -401,752 +1339,325 @@ def __init__(self, inputfilename,
+                    remove_all_traces(evt)  # remove all traces to save memory
+                if(evt_group_triggered):
+                    self._output_writer_hdf5.add_event_group(output_buffer)
 
-        # Create trigger structures if there are no triggering events.
-        # This is done to ensure that files with no triggering n_events
-        # merge properly.
-#         self._create_empty_multiple_triggers()
-
-        # save simulation run in hdf5 format (only triggered events)
-        t5 = time.time()
-        self._write_output_file()
         if self._outputfilenameNuRadioReco is not None:
-            self._eventWriter.end()
+            eventWriter.end()
             logger.debug("closing nur file")
 
-        if "simulation_mode" not in self._fin_attrs or self._fin_attrs['simulation_mode'] == "neutrino": # only calcualte Veff for neutrino simulations
-            try:
-                self.calculate_Veff()
-            except:
-                logger.error("error in calculating effective volume")
-
-        t_total = time.time() - t_start
-        outputTime = time.time() - t5
-
-        output_NuRadioRecoTime = "Timing of NuRadioReco modules \n"
-        ts = []
-        for iM, (name, instance, kwargs) in enumerate(self._evt.iter_modules(self._station.get_id())):
-            ts.append(instance.run.time[instance])
-        ttot = np.sum(np.array(ts))
-        for i, (name, instance, kwargs) in enumerate(self._evt.iter_modules(self._station.get_id())):
-            t = pretty_time_delta(ts[i])
-            trel = 100.*ts[i] / ttot
-            output_NuRadioRecoTime += f"{name}: {t} {trel:.1f}%\n"
-        logger.status(output_NuRadioRecoTime)
-
-        logger.status("{:d} events processed in {} = {:.2f}ms/event ({:.1f}% input, {:.1f}% ray tracing, {:.1f}% askaryan, {:.1f}% detector simulation, {:.1f}% output, {:.1f}% weights calculation)".format(self._n_showers,
-                                                                                         pretty_time_delta(t_total), 1.e3 * t_total / self._n_showers,
-                                                                                         100 * input_time / t_total,
-                                                                                         100 * (rayTracingTime - askaryan_time) / t_total,
-                                                                                         100 * askaryan_time / t_total,
-                                                                                         100 * detSimTime / t_total,
-                                                                                         100 * outputTime / t_total,
-                                                                                         100 * weightTime / t_total))
-        triggered = remove_duplicate_triggers(self._mout['triggered'], self._fin['event_group_ids'])
-        n_triggered = np.sum(triggered)
-        return n_triggered
```

Comment:
the new code does not return the number of triggered events.

Reply:
is this a problem?

Reply:
no not a big problem. my simulation script did use the return value. So we might break user code (in my case it was only to print it)